### ISSUE TITLE
rnn_crf模型训练时模型创建出错

### ISSUE THREAD

**Issue by lxwzafu** [shibing624/pycorrector#101]
环境：windows10 64位，VS2017，Python version=3.6.6，tensorflow version=1.13.1，pycorrector version=0.1.9，keras version=2.3.1
在测试pycorrector/rnn_crf模型时，运行train.py，程序执行到rnn_crf_model.py的create_model函数时出错：
![jt](https://user-images.githubusercontent.com/55593859/67397049-35287e80-f5db-11e9-8afb-e17371ef849b.jpg)
错误提示：Tensors in list passed to 'values' of 'ConcatV2' Op have types [bool, float32] that don't all match.
里面的变量值为：
len(word_dict)=1979
len(label_dict)=51
embedding_dim=100
rnn_hidden_dim=200
dropout=0.25
请问可能是什么原因？

**Comment by lxwzafu** [comment]
keras 版本问题，切换到2.2.0版本就好了。真是乱啊，一个版本一个版本的试，过程中会出现各种各样的问题。

**Comment by GC-Tom** [comment]
> 环境：windows10 64位，VS2017，Python version=3.6.6，tensorflow version=1.13.1，pycorrector version=0.1.9，keras version=2.3.1
> 在测试pycorrector/rnn_crf模型时，运行train.py，程序执行到rnn_crf_model.py的create_model函数时出错：
> ![jt](https://user-images.githubusercontent.com/55593859/67397049-35287e80-f5db-11e9-8afb-e17371ef849b.jpg)
> 错误提示：Tensors in list passed to 'values' of 'ConcatV2' Op have types [bool, float32] that don't all match.
> 里面的变量值为：
> len(word_dict)=1979
> len(label_dict)=51
> embedding_dim=100
> rnn_hidden_dim=200
> dropout=0.25
> 请问可能是什么原因？



> keras 版本问题，切换到2.2.0版本就好了。真是乱啊，一个版本一个版本的试，过程中会出现各种各样的问题。

请问训练结束之后 可以保存模型吗

**Comment by GC-Tom** [comment]
> keras 版本问题，切换到2.2.0版本就好了。真是乱啊，一个版本一个版本的试，过程中会出现各种各样的问题。

请问您训练结束后 可以保存模型吗

**Comment by shibing624** [comment]
可以

**Comment by lxwzafu** [comment]
可能是版本问题，运行过程中无val_acc值，只有crf_viterbi_accuracy值，所以这里需要监视crf_viterbi_accuracy才会保存模型。
checkpoint = ModelCheckpoint(save_model_path, monitor='crf_viterbi_accuracy', 
                                 save_best_only=True, mode='auto',
                                 save_weights_only=True, verbose=1)

**Comment by LZ798** [comment]
请问，你的RNN-CRF模型纠错改正效果好吗
